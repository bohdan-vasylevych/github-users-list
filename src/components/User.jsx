import React from 'react'
import {Link} from 'react-router-dom';
import {Col, Image, Panel} from 'react-bootstrap'

const User = ({avatar_url, html_url, login}) => (
    <Col md={3}>
        <Link to={`/${login}`} className='user'>
            <Panel>
                <Panel.Heading>
                    {login}
                </Panel.Heading>
                <Panel.Body>
                    <Image responsive src={avatar_url} alt={login} className='user-img'/>
                </Panel.Body>
                <Panel.Footer>
                    <p>{html_url}</p>
                </Panel.Footer>
            </Panel>
        </Link>
    </Col>
);

export default User;
