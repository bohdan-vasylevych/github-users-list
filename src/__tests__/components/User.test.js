import React from 'react';
import rendered from 'react-test-renderer';
import {BrowserRouter as Router} from 'react-router-dom';

import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import User from '../../components/User';

configure({adapter: new Adapter()});
const defaultProps = {
    avatar_url: 'avatar_url',
    html_url: 'html_url',
    login: 'login'
};

describe('User component', () => {

    test('matches the snapshot', () => {
        const tree = rendered.create(<Router><User {...defaultProps}/></Router>).toJSON();
        expect(tree).toMatchSnapshot();
    });

});
