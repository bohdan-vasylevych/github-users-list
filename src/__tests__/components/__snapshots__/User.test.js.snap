// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User component matches the snapshot 1`] = `
<div
  className="col-md-3"
>
  <a
    className="user"
    href="/login"
    onClick={[Function]}
  >
    <div
      className="panel panel-default"
    >
      <div
        className="panel-heading"
      >
        login
      </div>
      <div
        className="panel-body"
      >
        <img
          alt="login"
          className="user-img img-responsive"
          src="avatar_url"
        />
      </div>
      <div
        className="panel-footer"
      >
        <p>
          html_url
        </p>
      </div>
    </div>
  </a>
</div>
`;
